rai/node/rpc.cpp

```diff
@@ -2945,126 +3062,120 @@ void rai::rpc_handler::representatives_online ()
 
 void rai::rpc_handler::republish ()
 {
-	uint64_t count (1024U);
-	uint64_t sources (0);
-	uint64_t destinations (0);
-	boost::optional<std::string> count_text (request.get_optional<std::string> ("count"));
-	if (count_text.is_initialized ())
+	std::error_code ec;
+	auto count (decode_unsigned (request.get<std::string> ("count", "1024")));
+	if (!count)
 	{
-		auto error (decode_unsigned (count_text.get (), count));
-		if (error)
-		{
-			error_response (response, "Invalid count limit");
-		}
+		ec = error_rpc::invalid_count_limit;
 	}
-	boost::optional<std::string> sources_text (request.get_optional<std::string> ("sources"));
-	if (sources_text.is_initialized ())
+
+	auto sources (decode_unsigned (request.get<std::string> ("sources", "0")));
+	if (!ec && !sources)
 	{
-		auto sources_error (decode_unsigned (sources_text.get (), sources));
-		if (sources_error)
-		{
-			error_response (response, "Invalid sources number");
-		}
+		ec = error_rpc::invalid_sources_number;
 	}
-	boost::optional<std::string> destinations_text (request.get_optional<std::string> ("destinations"));
-	if (destinations_text.is_initialized ())
+
+	auto destinations (decode_unsigned (request.get<std::string> ("destinations", "0")));
+	if (!ec && !destinations)
```

Comment:
Nit: in an above function (I don't remember which) you don't check `ec` when parsing future values. It doesn't matter that much, but we should probably be consistent.

Reply:
Agreed, though I don't seem to find the missing check.